File: kuryer/lib/core/format.dart
import 'package:flutter/material.dart';

import 'theme.dart';

/// Mirrors `src/lib/format.ts` from the courier web app.

/// "123 456" — thousands separated by a space (ru-RU style).
String money(num n) {
  final s = n.round().abs().toString();
  final buf = StringBuffer();
  for (var i = 0; i < s.length; i++) {
    if (i > 0 && (s.length - i) % 3 == 0) buf.write(' ');
    buf.write(s[i]);
  }
  return (n < 0 ? '-' : '') + buf.toString();
}

String _two(int n) => n.toString().padLeft(2, '0');

/// "05.07 14:30"
String formatDateTime(String iso) {
  final d = DateTime.tryParse(iso)?.toLocal();
  if (d == null) return iso;
  return '${_two(d.day)}.${_two(d.month)} ${_two(d.hour)}:${_two(d.minute)}';
}

const _monthsShort = [
  'янв', 'фев', 'мар', 'апр', 'май', 'июн',
  'июл', 'авг', 'сен', 'окт', 'ноя', 'дек',
];

/// "05 июл"
String formatDay(String iso) {
  final d = DateTime.tryParse(iso)?.toLocal();
  if (d == null) return iso;
  return '${_two(d.day)} ${_monthsShort[d.month - 1]}';
}

/// Full timestamp used on the detail screen: "05.07.2026, 14:30:05"
String formatFull(String iso) {
  final d = DateTime.tryParse(iso)?.toLocal();
  if (d == null) return iso;
  return '${_two(d.day)}.${_two(d.month)}.${d.year}, '
      '${_two(d.hour)}:${_two(d.minute)}:${_two(d.second)}';
}

const statusLabelMap = {
  'pending': 'Yangi',
  'confirmed': 'Tasdiqlangan',
  'preparing': 'Tayyorlanmoqda',
  'ready': 'Tayyor',
  'accepted': 'Qabul qilindi',
  'delivering': 'Yetkazilmoqda',
  'delivered': 'Yetkazildi',
  'cancelled': 'Bekor qilindi',
};

String statusLabel(String s) => statusLabelMap[s] ?? s;

/// (background, foreground) colours for a status pill.
(Color, Color) statusPillColors(String s) {
  switch (s) {
    case 'confirmed':
      return (const Color(0xFFFEF3C7), const Color(0xFFB45309)); // amber
    case 'preparing':
      return (const Color(0xFFFFEDD5), const Color(0xFFC2410C)); // orange
    case 'ready':
      return (const Color(0xFFEDE9FE), const Color(0xFF6D28D9)); // violet
    case 'accepted':
      return (const Color(0xFFCFFAFE), const Color(0xFF0E7490)); // cyan
    case 'delivering':
      return (const Color(0xFFDBEAFE), const Color(0xFF1D4ED8)); // blue
    case 'delivered':
      return (const Color(0xFFD1FAE5), const Color(0xFF047857)); // emerald
    case 'cancelled':
      return (const Color(0xFFFFE4E6), const Color(0xFFBE123C)); // rose
    case 'pending':
    default:
      return (AppColors.slate100, AppColors.slate700);
  }
}

const _paymentLabelMap = {
  'cash': 'Naqd',
  'payme': 'Payme',
  'click': 'Click',
  'uzum': 'Uzum',
};

String paymentLabel(String? m) => m == null ? '—' : (_paymentLabelMap[m] ?? m);

/// Yandex Maps navigation link (when lat/lng present).
String? mapsUrl(double? lat, double? lng) => (lat != null && lng != null)
    ? 'https://yandex.com/maps/?rtext=~$lat,$lng&rtt=auto'
    : null;

/// "1 kg", "3 dona"
String qtyUnit(num quantity, String? unit) {
  final q = quantity == quantity.roundToDouble()
      ? quantity.round().toString()
      : quantity.toString();
  return '$q ${(unit == null || unit.isEmpty) ? 'dona' : unit}';
}

/// "4.2 km" / "850 m"
String? distanceLabel(double? km) {
  if (km == null) return null;
  return km < 1 ? '${(km * 1000).round()} m' : '${km.toStringAsFixed(1)} km';
}

/// "~25 daqiqa"
String? etaLabel(int? minutes) =>
    (minutes != null && minutes > 0) ? '~$minutes daqiqa' : null;

File: courier/src/lib/format.ts
import type { OrderStatus, PaymentMethod } from "../types";

export const money = (n: number) =>
  n.toLocaleString("ru-RU").replace(/,/g, " ");

export const formatDateTime = (iso: string) =>
  new Date(iso).toLocaleString("ru-RU", {
    day: "2-digit",
    month: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
  });

export const formatDay = (iso: string) =>
  new Date(iso).toLocaleDateString("ru-RU", { day: "2-digit", month: "short" });

export const STATUS_LABEL: Record<string, string> = {
  pending: "Yangi",
  confirmed: "Tasdiqlangan",
  preparing: "Tayyorlanmoqda",
  ready: "Tayyor",
  accepted: "Qabul qilindi",
  delivering: "Yetkazilmoqda",
  delivered: "Yetkazildi",
  cancelled: "Bekor qilindi",
};

export const STATUS_PILL: Record<string, string> = {
  pending: "bg-slate-100 text-slate-700",
  confirmed: "bg-amber-100 text-amber-700",
  preparing: "bg-orange-100 text-orange-700",
  ready: "bg-violet-100 text-violet-700",
  accepted: "bg-cyan-100 text-cyan-700",
  delivering: "bg-blue-100 text-blue-700",
  delivered: "bg-emerald-100 text-emerald-700",
  cancelled: "bg-rose-100 text-rose-700",
};

export const statusLabel = (s: OrderStatus) => STATUS_LABEL[s] ?? s;
export const statusPill = (s: OrderStatus) =>
  STATUS_PILL[s] ?? "bg-slate-100 text-slate-600";

export const PAYMENT_LABEL: Record<string, string> = {
  cash: "Naqd",
  payme: "Payme",
  click: "Click",
  uzum: "Uzum",
};

export const paymentLabel = (m?: PaymentMethod) =>
  m ? PAYMENT_LABEL[m] ?? m : "—";

/** Google Maps navigatsiya havolasi (lat/lng bo'lsa). */
export const mapsUrl = (lat?: number | null, lng?: number | null) =>
  lat != null && lng != null
    ? `https://www.google.com/maps/dir/?api=1&destination=${lat},${lng}`
    : null;

/** Miqdor + o'lchov birligi: "1 kg", "3 dona". */
export const qtyUnit = (quantity: number, unit?: string | null) =>
  `${quantity} ${unit || "dona"}`;

/** Masofa: "4.2 km" / "850 m". */
export const distanceLabel = (km?: number | null) => {
  if (km == null) return null;
  return km < 1 ? `${Math.round(km * 1000)} m` : `${km.toFixed(1)} km`;
};

/** ETA: "~25 daqiqa". */
export const etaLabel = (minutes?: number | null) =>
  minutes ? `~${minutes} daqiqa` : null;
